app/questionnaire/placeholder_parser.py

```diff
@@ -74,8 +76,8 @@ def __init__(
         self._routing_path_block_ids_by_section_key: dict = {}
 
     def __call__(
-        self, placeholder_list: Sequence[Mapping]
-    ) -> MutableMapping[str, Union[ValueSourceEscapedTypes, ValueSourceTypes]]:
+        self, placeholder_list: Sequence[Mapping[str, Any]]
```

Comment:
My opinion is if it is deterministic, this assign the types. So `dict[str, str]` but avoid `dict[str, str | list]` because that mean any key could be a list or str which isn't helpful. I would say:
1. Explicitly type if the values have the same type, i.e `dict[str, str]`
2. If the value can be multiple types, but the keys are deterministic and is not a schema partial, we should use `TypedDict`.
3. All else, just generics like `dict`.

Thoughts?

Reply:
I'm happy with that approach, I think thats a sensible choice so will go with that

Reply:
I think that approach sounds good, I'll update my PR's Mapping[str, Any] instances to Mapping (same with MutableMappings).

Reply:
I've done the same, removed any [str, Any]s I've added alongside a couple of others in the questionnaire module just for consistency with my changes but I've left any outside of that module

Reply:
I haven't got a huge preference either way to be honest. I do see your point but I can also see that to me `Mapping[str, Any]` is a bit more explicit and adds a bit more explanation even if it is only a small amount

Reply:
Happy with this approach 👍

Reply:
Had a read through, sounds good 👍 . Should we update our type hinting gudie to reference this ?

Reply:
`[str, Any]` doesn't bring much value imho, but we do Mappings with sets as keys etc. so it is not obvious it will always be `str` or `int`.

Reply:
If the value is the same for all keys then we would still type it, regardless of it if is set etc.

Reply:
@Yuyuutsu Good shout. Looks like @kylelawsonAND has done it in his PR 👍